packages/vite/src/node/server/middlewares/static.ts

```diff
@@ -109,13 +109,18 @@ export function serveStaticMiddleware(
     }
 
     if (redirected) {
-      req.url = redirected
+      req.url = encodePercent(redirected)
```

Comment:
I did this for understandability and perf. But I changed to `encodeURIComponent` because now I think it is more understandable to use `encodeURIComponent` and it turned out it didn't have much perf affect.

[perf](https://perf.link/#eyJpZCI6ImRocnlrOTA5OGI4IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IHVybCA9IFwiLyU0MGFhL1wiXG5jb25zdCB1cmwyID0gXCIvYWEvXCIiLCJ0ZXN0cyI6W3sibmFtZSI6ImVuY29kZVVSSUNvbXBvbmVudElmTmVlZGVkIiwiY29kZSI6InVybC5pbmNsdWRlcygnJScpID8gZW5jb2RlVVJJQ29tcG9uZW50KHVybCkgOiB1cmw7XG51cmwyLmluY2x1ZGVzKCclJykgPyBlbmNvZGVVUklDb21wb25lbnQodXJsMikgOiB1cmwyOyIsInJ1bnMiOlsyNjgxMDAwLDE2MDAwLDI1ODgwMDAsMjIxOTAwMCwxMzcxMDAwLDY2MjAwMCwxODAwMCwyMTY2MDAwLDI1MDkwMDAsMTk1OTAwMCwxMjI5MDAwLDE3MzgwMDAsMTQ4NzAwMCwyMzA4MDAwLDE0MDQwMDAsMTcxMDAwLDE3MTEwMDAsMjAxMjAwMCwyNDczMDAwLDEwMDAsMTU3NzAwMCwxODk1MDAwLDIzMTQwMDAsMTIyMjAwMCwyNjgwMDAwLDEyNjEwMDAsOTIyMDAwLDE5NTgwMDAsMTk1OTAwMCwyMjk4MDAwLDE3MzYwMDAsMjYyNTAwMCwxMzAxMDAwLDIyNTAwMDAsMjU1MDAwMCwxNTE0MDAwLDM3MDAwLDE2NDcwMDAsOTcwMDAsMTMxMjAwMCwyNzMxMDAwLDI3NzEwMDAsMjU3OTAwMCwxODc1MDAwLDQzMDAwMCwyNzU4MDAwLDg5NzAwMCwyMDU3MDAwLDEwNjAwMDAsMjA4OTAwMCwyNjcyMDAwLDIzNTEwMDAsMjU1NTAwMCw4NDgwMDAsMTk4MjAwMCwyNzA2MDAwLDI4NjIwMDAsMTYzNzAwMCwyNTc2MDAwLDE1OTMwMDAsMTY2NDAwMCwxMTk5MDAwLDI5MTAwMCwyMTkwMDAsMTcyMjAwMCwxNDI5MDAwLDE1MTgwMDAsMjA2NzAwMCwyMTc0MDAwLDI4MDkwMDAsMjAzNjAwMCwyMjU0MDAwLDE2MjEwMDAsMTg5NDAwMCwxODkwMDAwLDI3NTMwMDAsMTU5NDAwMCwxNjM4MDAwLDI4NDMwMDAsMjQ1MzAwMCw5MzkwMDAsMTk0OTAwMCwyNTEyMDAwLDExMjAwMDAsMjU3MDAwMCwxMjY3MDAwLDE3NzgwMDAsMjI3MjAwMCw0ODgwMDAsMjM0NzAwMCwxNDk2MDAwLDQ3MDAwMCwyMzQyMDAwLDI0MTkwMDAsMTIwMjAwMCwyNDA1MDAwLDgwMDAsODQzMDAwLDU2MDAwLDE1OTUwMDBdLCJvcHMiOjE3MTA1MzB9LHsibmFtZSI6ImVuY29kZVVSSUNvbXBvbmVudCIsImNvZGUiOiJlbmNvZGVVUklDb21wb25lbnQodXJsKTtcbmVuY29kZVVSSUNvbXBvbmVudCh1cmwyKTsiLCJydW5zIjpbMTA4NTAwMCwyMzc0MDAwLDEzNTkwMDAsMTM5NDAwMCwxMjUwMDAwLDE0NzYwMDAsMjcwMTAwMCwxODkwMDAwLDEzODUwMDAsNzc3MDAwLDEwNzAwMCw4OTYwMDAsMjY0MzAwMCw0OTgwMDAsNTYwMDAwLDExODMwMDAsMjYwMDAsNjUyMDAwLDg0MjAwMCwxMjQwMDAwLDE1MDEwMDAsNDU2MDAwLDU5NjAwMCwyMzI1MDAwLDEyOTgwMDAsODQ2MDAwLDE5MjgwMDAsMjM3MjAwMCwyMzU5MDAwLDExMzQwMDAsMTk3NTAwMCwyMzY0MDAwLDIyMzcwMDAsMTIzMDAwMCwyMDUyMDAwLDQzMzAwMCwxMDk0MDAwLDE5ODAwMCwxNDk3MDAwLDExMDAwLDE0NTAwMDAsMTI5OTAwMCwyNjY1MDAwLDE2NzUwMDAsMjM2MjAwMCwxMTc1MDAwLDIzODIwMDAsNzczMDAwLDIzMTEwMDAsMTE1ODAwMCwyNTkxMDAwLDEwNDAwMDAsMTA4NTAwMCwxNjAzMDAwLDQ0NTAwMCwxMTI3MDAwLDEzMTAwMDAsNDA4MDAwLDI1NzEwMDAsMTM1MDAwLDE2MjAwMCw1MDAwMDAsMTE5ODAwMCwyMDY4MDAwLDcyNDAwMCwyNjEwMDAsNDQwMDAsMjUzMDAwLDExNjAwMDAsMTkyNDAwMCwyNjExMDAwLDEzMDgwMDAsNjMzMDAwLDk5MDAwMCw3MDgwMDAsMTU3NDAwMCwyMjE4MDAwLDc5MDAwLDE3OTUwMDAsMTM5NzAwMCw1NjAwMDAsNzY0MDAwLDEyMzIwMDAsMjU4NTAwMCw5NjMwMDAsMTQ2MDAwLDE1NjQwMDAsMzQ1MDAwLDg3NTAwMCwyMTIwMDAsNDc4MDAwLDI1NDIwMDAsMjQ3MzAwMCwxMzUwMDAwLDI0ODgwMDAsMTQwNDAwMCwxNTAxMDAwLDIwMjUwMDAsNjUyMDAwLDczOTAwMF0sIm9wcyI6MTI4MzE0MH0seyJuYW1lIjoiZW5jb2RlUGVyY2VudCIsImNvZGUiOiJ1cmwuaW5jbHVkZXMoJyUnKSA%2FIHVybC5yZXBsYWNlKC8lL2csICclMjUnKSA6IHVybDtcbnVybDIuaW5jbHVkZXMoJyUnKSA%2FIHVybDIucmVwbGFjZSgvJS9nLCAnJTI1JykgOiB1cmwyOyIsInJ1bnMiOlsxNDkwMDAsMjA0MTAwMCwyNzIwMDAsNjY2MDAwLDQzMjAwMCwxNzU3MDAwLDIwOTkwMDAsMjM0MzAwMCwxNzMwMDAsMTQ2MDAwLDIyOTUwMDAsMTExODAwMCwyMDI4MDAwLDE3MDQwMDAsMTQxMzAwMCwxMDMyMDAwLDExNDAwMDAsMTg4NjAwMCwyMzc4MDAwLDI1MzEwMDAsMjQ2NTAwMCwyMDQxMDAwLDIzNzYwMDAsMTM2MDAwMCw2NjAwMCw4MzMwMDAsODMwMDAsOTkzMDAwLDE1MzQwMDAsMTkwMDAwLDI1NTAwMCwxNTI3MDAwLDE0MjcwMDAsMTcwMDAsNzUzMDAwLDEyMjAwMCwyNTI1MDAwLDIxMjUwMDAsMTk2OTAwMCw5NzQwMDAsMjQ0MjAwMCwyNDgxMDAwLDIzMjIwMDAsMzQwMDAwLDkyNDAwMCwyOTcwMDAsMjI5NzAwMCwyMjg3MDAwLDE2NjAwMDAsMjM4MTAwMCwyMzgwMDAwLDE2MzgwMDAsMTk4MDAwLDEyNjgwMDAsMTkzNzAwMCwxNDI1MDAwLDQ3NjAwMCwxNDcyMDAwLDIwNTYwMDAsMjE1ODAwMCwxODcyMDAwLDIxMDEwMDAsMjAyNzAwMCwxOTYyMDAwLDkyNTAwMCwxMTg3MDAwLDE4NTEwMDAsMTYwNjAwMCwyNjgwMDAsNTc2MDAwLDg1MzAwMCwyMTE2MDAwLDExNDMwMDAsMTkxNTAwMCwxMTE4MDAwLDk1MTAwMCwyOTEwMDAsMTU3NjAwMCwyODcwMDAsMTAzNDAwMCw4NDEwMDAsMTQyOTAwMCw2MTkwMDAsMTMxNTAwMCwyMjQwMDAwLDcyMjAwMCw3MDMwMDAsMTE1NjAwMCwyMzE4MDAwLDE3NTYwMDAsMTk0NjAwMCwxMTE0MDAwLDEzODcwMDAsNDQ5MDAwLDEyNTIwMDAsMjI2NzAwMCw0MTkwMDAsNjc5MDAwLDE3MTkwMDAsMTU4NjAwMF0sIm9wcyI6MTM1MjUzMH1dLCJ1cGRhdGVkIjoiMjAyMi0wNy0wOFQwNDowNzo0OS45NzVaIn0%3D)